lib/screens/v2/candidate_generator/dup/alerts.ex

```diff
@@ -127,14 +142,195 @@ defmodule Screens.V2.CandidateGenerator.Dup.Alerts do
   # Otherwise, returns the alerts unchanged.
   @spec alert_special_cases(list(Alert.t()), Screens.Config.Screen.t()) ::
           {:normal, list(Alert.t())} | {:special, list(Screens.V2.WidgetInstance.t())}
-  defp alert_special_cases(alerts, _config) do
-    # Special cases go here!
-    # Use the DupSpecialCaseAlert widget to directly specify the serialized data.
+  defp alert_special_cases([], _), do: {:normal, []}
+
+  defp alert_special_cases(
+         alerts,
+         %Screen{app_params: %Dup{alerts: %AlertsConfig{stop_id: stop_id}}}
+       ) do
+    cond do
+      stop_id === "place-kencl" -> kenmore_special_case(alerts)
+      stop_id === "place-wtcst" -> wtc_special_case(alerts)
+      # precise case where there is an SL alert detouring through a particular location
+      # -> {:special, widgets}
+      true -> {:normal, alerts}
+    end
+  end
+
+  # In the case where Kenmore has 2 or more boundary shuttles / suspensions to the west,
+  # don't only select 1 alert; instead look at all alerts and make custom text
+  @spec kenmore_special_case(list(Alert.t())) :: {:special, list(Screens.V2.WidgetInstance.t())}
+  def kenmore_special_case(alerts) do
+    branches =
+      alerts
+      |> Enum.filter(fn a -> a.effect === :shuttle end)
+      |> Enum.map(
+        &check_entities_for_stops(&1, [
+          %{branch: "b", stop: "70149"},
+          %{branch: "c", stop: "70211"},
+          %{branch: "d", stop: "70187"}
+        ])
+      )
+      |> Enum.sort()
+      |> Enum.uniq()
+
+    if length(branches) > 1 do
+      alert_ids = Enum.map(alerts, fn a -> a.id end)
+
+      {:special,
+       [
+         %DupSpecialCaseAlert{
+           alert_ids: alert_ids,
+           serialize_map: %{
+             text: %Screens.Config.V2.FreeTextLine{
+               icon: :warning,
+               text: get_kenmore_special_text(branches, :partial_alert)
+             },
+             color: :green
+           },
```

Comment:
My gut tells me this is the wrong place for this bit. I get these are special cases, but putting `serialize` logic in the `CandidateGenerator` breaks the mold a bit on how we prepare `WidgetInstance`s. Is it possible to allow `DupSpecialCaseAlert.serialize` to handle this?

Reply:
I'm waffling back and forth and not sure how I feel. I went in this direction after reading @jzimbel-mbta's comment in his code `# Use the DupSpecialCaseAlert widget to directly specify the serialized data.` And it kinda makes sense that we want to break the mold in this case, because we go through all these conditionals in the `CandidateGenerator` to determine yes, we are working with a special case here, that to then pass this info to the `WidgetInstance` and then have to go through pattern matching to again say yes! this is a special case, seems a little redundant.

But if it's much cleaner to put the serialization in the WidgetInstance, then I understand breaking the mold might not be worth it

Reply:
That's a good point. We've had edge cases before, but this one is kinda the leader of edge cases. So makes sense that it needs to stray from convention.

Reply:
Answered in slack, but in short: **Yes**, I think it's reasonable to deviate from the approach I had prescribed without really looking into how complicated some of these special cases are!

Reply:
I'm not worried about deviating from the originally prescribed solution, but trying to figure out which organization is actually better

Reply:
For now, I'll reorganize and then get your weigh-in